### PR TITLE
Source Iterable: Update iterable.md docs

### LINKED ISSUE #7735: Source Iterable: 401 Client Error - Unauthorized for url
## Enviroment
OS Version / Instance: Linux EC2 m5.2xlarge
Deployment: Docker
Airbyte Version: 0.30.20-alpha
Source name: Iterable 0.1.9
destination: Redshift 0.3.19

 Trying iterable for the first time. Does not work. Is it because this is a read_only API key?
 
## Current Behavior
401 Client Error: Unauthorized for url: https://api.iterable.com/api//export/data.json....
Also API key is being output to logs! (edited) 

## Expected Behavior
*Tell us what should happen.*

## Logs
[logs-10916-0.txt](https://github.com/airbytehq/airbyte/files/7496898/logs-10916-0.txt)

## Slack Convo
https://airbytehq.slack.com/archives/C01MFR03D5W/p1635482271308800

## Acceptance Criteria
1. Verify that a read_only key is sufficient for reading all streams. If it is not, then change the documentation to indicate this
2. Ensure that the API key is never logged anywhere like mentioned in the Slack conversation

### PR DESCRIPTION
**Pull request by htrueman**
## What
The issue appeared in https://github.com/airbytehq/airbyte/issues/7735 

## How
Update docs

## 🚨 User Impact 🚨
Clears the API key usage a bit.

### CODE REVIEW

File: docs/integrations/sources/iterable.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ The Iterable connector should not run into Iterable API limitations under normal
 ### Requirements
 
 * Iterable Account
-* Iterable API Key
+* Iterable API Key with `standard` permissions. See https://support.iterable.com/hc/en-us/articles/360043464871-API-Keys- for more details.
```
Comment (Zirochkaa) [inline docs/integrations/sources/iterable.md:51]:
```suggestion
* Iterable API Key with `standard` permissions. See [API Keys docs](https://support.iterable.com/hc/en-us/articles/360043464871-API-Keys-) for more details.
```

**Review by Zirochkaa — APPROVED** [review]

File: docs/integrations/sources/iterable.md (hunk 1 above)
Comment (sherifnada) [inline docs/integrations/sources/iterable.md:51]: @htrueman shouldn't this be read-only like detailed by the user in the linked issue?

Comment (htrueman) [reply docs/integrations/sources/iterable.md:51]: Hm, but he told that he used `standard` permissions to make it work. See https://github.com/airbytehq/airbyte/issues/7735#issuecomment-978882344

Comment (sherifnada) [reply docs/integrations/sources/iterable.md]: nevermind I am blind. you are correct - ignore me!

**Review by sherifnada — APPROVED** [review]

**Pull request merged**